File: Simple_Workshop_Projects/JavaProject/app/src/main/java/gr/sfhmmy/java/workshop/MainActivity.java
package gr.sfhmmy.java.workshop;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import gr.sfhmmy.java.workshop.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding layoutAdapter;
    MainActivityViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        viewModel = new ViewModelProvider(this, new ViewModelProvider.NewInstanceFactory())
                .get(MainActivityViewModel.class);

        layoutAdapter = DataBindingUtil.setContentView(this, R.layout.activity_main);
    }

    public void showGreeting(View view) {
        int age = Integer.parseInt(layoutAdapter.ageEditText.getText().toString());
        layoutAdapter.greetingOutputTextView.setText(viewModel.calculatingGreeting(age, layoutAdapter.nameEditText.getText().toString()));
    }
}

File: Simple_Workshop_Projects/KotlinProject/app/src/main/java/gr/sfhmmy/kotlin/workshop/MainActivity.kt
package gr.sfhmmy.kotlin.workshop

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import gr.sfhmmy.kotlin.workshop.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    var layoutAdapter: ActivityMainBinding? = null
    var viewModel: MainActivityViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(this, NewInstanceFactory())
                .get(MainActivityViewModel::class.java)
        layoutAdapter = DataBindingUtil.setContentView(this, R.layout.activity_main)
    }

    fun showGreeting(view: View?) {
        val age = layoutAdapter?.ageEditText?.text.toString().toInt()
        layoutAdapter?.greetingOutputTextView?.text = viewModel?.calculatingGreeting(age, layoutAdapter?.nameEditText?.text.toString())
    }
}